test/e2e/utils/recordedit-helpers.js

```diff
@@ -317,62 +321,78 @@ exports.testPresentationAndBasicValidation = function(tableParams, isEditMode) {
-                            var livePreviewLink = element.all(by.className('live-preview'));
-                            for (i = 0; i < markdownTestParams.length; i++) {
-                                chaisePage.recordEditPage.clearInput(markdownField);
-                                browser.sleep(10);
-                                
-                                (function(input, markdownOut) {
-                                    markdownField.sendKeys(input);
-                                    livePreviewLink.click();
-                                    let mdDiv = element(by.css('[ng-bind-html="ctrl.params.markdownOut"]'));
-                                    browser.wait(EC.presenceOf(mdDiv), browser.params.defaultTimeout);
-                                    expect(mdDiv.getAttribute('outerHTML')).toContain(markdownOut, "Error during markdown generation");
-                                    element(by.className('modal-close')).click();
-                                })(markdownTestParams[i].raw, markdownTestParams[i].markdown);
-                            } //for
-                        });
-                    }).pend("should be changed because the live-preview selector is not unique, and also it will changed");
-                    
+                      //Both preview is being tested.
+                      markdownCols.forEach(function(descCol) {
+                        var markdownField = chaisePage.recordEditPage.getInputById(recordIndex, descCol.title);
+                        btnIndex = (recordIndex * 2) + 1;
+                        var PrevBtn = element.all(by.css('button[title="Preview"]')).get(btnIndex);       //test inline preview
+                        var modalPrevBtn = element.all(by.css('button[title="Fullscreen Preview"]')).get(btnIndex);       //test modal preview
+                        var ctrlA = protractor.Key.chord(protractor.Key.CONTROL, "a");
+                        for (i = 0; i < markdownTestParams.length; i++) {
+                          markdownField.clear();
+                          (function(input, markdownOut, comm, btnIdx) {
+
+                                markdownField.sendKeys(input,ctrlA);
+                                if(comm !=' ')
+                                { //If keyboard shortcut found for markdown elements then send click command.
+                                        let v = "button[data-hotkey='"+comm+"']";
+                                        element.all(by.css(v)).get(btnIdx).click();
+                                }
+
+                                modalPrevBtn.click();
+                                let mdDiv = element(by.css('[ng-bind-html="ctrl.params.markdownOut"]'));
+                                browser.wait(EC.presenceOf(mdDiv), browser.params.defaultTimeout);
+                                expect(mdDiv.getAttribute('outerHTML')).toContain(markdownOut, "Error during markdown pop-up generation");
```

Comment:
You can wrap these error messages in the helper function that I wrote in this function to make it easier to track where it exactly failed. `colError(descCol.name, "Error during markdown pop-up generation")`. It will just provide the column name and row index that this error occurred in it.